app/v2/models/errors/CommonMtdErrors.scala

```diff
@@ -16,178 +16,157 @@
+      httpStatus = BAD_REQUEST
     )
 
 object RuleIncorrectOrEmptyBodyError
     extends MtdError(
       code = "RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED",
-      message = "An empty or non-matching body was submitted"
-    )
-
-object RuleAlreadySubmittedError
-    extends MtdError(
-      code = "RULE_ALREADY_SUBMITTED",
-      message = "An End of Period Statement already exists for this business' accounting period"
-    )
-
-object RuleEarlySubmissionError
-    extends MtdError(
-      code = "RULE_EARLY_SUBMISSION",
-      message = "An End Of Period Statement cannot be submitted before the end of the accounting period"
-    )
-
-object RuleLateSubmissionError
-    extends MtdError(
-      code = "RULE_LATE_SUBMISSION",
-      message = "The period to finalise has passed"
-    )
-
-object RuleNonMatchingPeriodError
-    extends MtdError(
-      code = "RULE_NON_MATCHING_PERIOD",
-      message = "An End of Period Statement without a matching accounting period cannot be submitted"
-    )
-
-object RuleBusinessValidationFailure {
-  val code = "RULE_BUSINESS_VALIDATION_FAILURE"
-
-  def apply(message: String, errorId: String): MtdError =
-    MtdError(code = code, message = message, errorId = Some(errorId))
-}
-
-object RuleBusinessValidationFailureTys
-    extends MtdError(
-      code = "RULE_BUSINESS_VALIDATION_FAILURE",
-      message = "There are business validation rule failures."
+      message = "An empty or non-matching body was submitted",
+      httpStatus = BAD_REQUEST
     )
 
 //Standard Errors
 object NotFoundError
     extends MtdError(
       code = "MATCHING_RESOURCE_NOT_FOUND",
-      message = "Matching resource not found"
+      message = "Matching resource not found",
+      httpStatus = NOT_FOUND
     )
 
 object InternalError
     extends MtdError(
       code = "INTERNAL_SERVER_ERROR",
-      message = "An internal server error occurred"
+      message = "An internal server error occurred",
+      httpStatus = INTERNAL_SERVER_ERROR
     )
 
 object BadRequestError
     extends MtdError(
       code = "INVALID_REQUEST",
-      message = "Invalid request"
+      message = "Invalid request",
+      httpStatus = BAD_REQUEST
     )
 
 object ServiceUnavailableError
     extends MtdError(
       code = "SERVICE_UNAVAILABLE",
-      message = "Internal server error"
+      message = "Internal server error",
+      httpStatus = INTERNAL_SERVER_ERROR
+    )
+
+object InvalidHttpMethodError
+    extends MtdError(
+      code = "INVALID_HTTP_METHOD",
+      message = "Invalid HTTP method",
+      httpStatus = METHOD_NOT_ALLOWED
     )
 
 //Authorisation Errors
-object UnauthorisedError
+object ClientNotAuthenticatedError
+    extends MtdError(
+      code = "CLIENT_OR_AGENT_NOT_AUTHORISED",
+      message = "The client and/or agent is not authorised",
+      httpStatus = UNAUTHORIZED
+    )
+
+/** Authentication OK but not allowed access to the requested resource
```

Comment:
Would it be worth putting this comment in all the other APIs? Makes it so much clearer to me the difference between them... is it worth doing so?

Reply:
This wasn't my comment so it is already in some APIs. But yes, makes sense if it helps.